src/dbt_osmosis/core/column_level_knowledge_propagator.py

```diff
@@ -36,10 +40,34 @@ def _build_node_ancestor_tree(
     return family_tree
 
 
+def _get_member_yaml(member):
```

Comment:
Lets add typing for this. We can use Any for the input type if you want since the if/else guarantees a known type. Otherwise use a union type. And the return type should be Optional[dict]

Reply:
Shouldn't input be `ManifestNode` since that's [what member is](https://github.com/z3z1ma/dbt-osmosis/blob/main/src/dbt_osmosis/core/column_level_knowledge_propagator.py#L50)

Reply:
Yeah for sure. The if/elif/else at the beginning serves as a type guard and guarantees the type from a type checker perspective in the function body. But yes ManifestNode is more correct and will catch a bug earlier 👍